Jenkins slave erlang.22 elixir.1.12.3

R-Jellema opened the pull request:
Dockerfile voor elixir/erlang agent vanuit officiële erlang image.
Container draait op debian buster 10.
Container zal wel moeten uitgevoerd met de param de "-dit" param. Anders sluit de container zichzelf direct af.

R-Jellema committed:
Create Dockerfile

R-Jellema committed:
Added sonarscanner properties.

rodymiddelkoop reviewed (changes requested):
Met deze Dockerfile kun je wel Erlang/Elixir projecten builden maar om 'm als Jenkins slave te gebruiken missen de volgende instructies die je op het einde kunt toevoegen:

# Define working directory.
WORKDIR /data

# Set user jenkins to the image
RUN useradd -m -d /home/jenkins -s /bin/sh jenkins &&\
    echo "jenkins:jenkins" | chpasswd
RUN apt-get update && apt-get install -y git

#JDK15 for connecting Jenkins master and slave using JNLP
RUN \
  apt-get -q update && \
  apt-get install curl -y && \
  cd usr/lib/jvm &&  curl -O https://download.java.net/java/GA/jdk15.0.1/51f4f36ad4ef43e39d0dfdbaf6549e32/9/GPL/openjdk-15.0.1_linux-x64_bin.tar.gz && \
  tar -xvzf openjdk-15.0.1_linux-x64_bin.tar.gz && \
  rm -rf openjdk-15.0.1_linux-x64_bin.tar.gz


# Sonar Scanner
RUN apt-get update && apt-get install -y unzip wget bzip2 && wget https://binaries.sonarsource.com/Distribution/sonar-scanner-cli/sonar-scanner-cli-4.5.0.2216.zip --quiet && unzip sonar-scanner-cli-4.5.0.2216.zip -d /opt

COPY "sonar-scanner.properties" /opt/sonar-scanner-4.5.0.2216/conf

# Make Java15 the default
RUN update-alternatives  --install /usr/bin/java java /usr/lib/jvm/jdk-15.0.1/bin/java 1000 && update-alternatives  --install /usr/bin/javac javac /usr/lib/jvm/jdk-15.0.1/bin/javac 1001
RUN update-alternatives --set java /usr/lib/jvm/jdk-15.0.1/bin/java && update-alternatives --set javac /usr/lib/jvm/jdk-15.0.1/bin/javac

# Standard SSH port
EXPOSE 22

# Default command
CMD ["/usr/sbin/sshd", "-D"]

R-Jellema commented:
Excuus! wordt geregeld!

R-Jellema committed:
Updated dockerfile to work with Jenkins.

Added OpenSSH and configured auto start up.
Defined Jenkins workspace.
Configured Jenkins user for the image.
JDK 15 for connecting Jenkins master and slave using JNLP and for Sonar.
Configured sonar scanner.
JDK 15 set as default.
expose ssh port 22 for Jenkins master.
Default command added.
